keras/layers/regularization/dropout.py

```diff
@@ -49,10 +49,8 @@ def __init__(self, rate, noise_shape=None, seed=None, **kwargs):
         self.rate = rate
         self.seed = seed
         self.noise_shape = noise_shape
-        if rate > 0:
-            self.seed_generator = backend.random.SeedGenerator(seed)
+        self.seed_generator = backend.random.SeedGenerator(seed)
```

Comment:
I think we should always initialize a `self.seed_generator` regardless of how `rate` is set.
Otherwise, the user won't be able to reactivate the dropout layer by setting a new `rate` and recompiling the training graph.

Reply:
The rate is not intended to be settable. If we want to make it settable we should make a setter that would create the state variable (after unlocking the layer).

Reply:
Got it. I have also applied this idea to `AlphaDropout`, `GaussianDropout` and `GaussianNoise`